runtime/src/bank.rs

```diff
@@ -1206,9 +1211,22 @@ impl Bank {
     }
 
     pub fn new_for_tests(genesis_config: &GenesisConfig) -> Self {
+        Self::_new_for_tests(genesis_config, &None)
+    }
+
+    pub fn new_for_tests_with_config(genesis_config: &GenesisConfig, test_config: &Option<BankTestConfig>) -> Self {
+        Self::_new_for_tests(genesis_config, test_config)
+    }
+
+    fn _new_for_tests(genesis_config: &GenesisConfig, test_config: &Option<BankTestConfig>) -> Self {
```

Comment:
I don't see any reason for this additional internal method.

Incidentally, I think it's going to be pretty confusing to have a `Bank::new_with_config_for_tests()` and a `Bank::new_for_tests_with_config()`, one of which calls the other.

Reply:
I don't think this has been resolved. I still see an unnecessary internal method. And I'm still concerned about the confusingly similar method names.

Reply:
I guess I was confused how to resolve this given a config is needed and trent and you agreed on this. So I understood that meaning I should keep it. Are you saying the default interface should be the one with the config? Theres no way to do what I'm doing without the added argument, if I want to keep the old interface the same I need the shim, or I need to change the default API to add the new argument.

Reply:
I am saying that the separate internal `Bank::_new_for_tests()` method is unnecessary. You can have `Bank::new_for_tests()` call `Bank::new_for_tests_with_config()` and put the internal logic in there.

Reply:
Ahhh ok gotcha.

Reply:
Addressed.